services/scraper/scraper.go

```diff
@@ -0,0 +1,127 @@
+package scraper
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hoangmirs/go-scraper/models"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/gocolly/colly/v2"
+	"github.com/gocolly/colly/v2/extensions"
```

Comment:
How about renaming the import here to avoit confusion:

```suggestion
	collyExtensions "github.com/gocolly/colly/v2/extensions"
```

WHen reading later `extensions.RandomUserAgent(service.Collector)`, I was confused where this came from 🙃